runtime/src/main/java/org/corfudb/runtime/view/AddressSpaceView.java

```diff
@@ -114,15 +117,30 @@ public AddressSpaceView(@Nonnull final CorfuRuntime runtime) {
                 .recordStats()
                 .build();
 
-        Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
         MicroMeterUtils.gauge("address_space.read_cache.hit_ratio", readCache, cache -> cache.stats().hitRate());
+        MicroMeterUtils.gauge("address_space.read_cache.estimated_avg_entry_size", calculateEstimatedAvgEntrySize());
+        MicroMeterUtils.gauge("address_space.read_cache.cache_size", readCache, cache -> cache.size());
+        Optional<MeterRegistry> metricsRegistry = MeterRegistryProvider.getInstance();
         metricsRegistry.map(registry -> GuavaCacheMetrics.monitor(registry, readCache, "address_space.read_cache"));
     }
 
     private void handleEviction(RemovalNotification<Long, ILogData> notification) {
         if (log.isTraceEnabled()) {
             log.trace("handleEviction: evicting {} cause {}", notification.getKey(), notification.getCause());
         }
+        updateCurrentDataSizeInCache(currentDataSizeInCache.get(),
+                (notification.getValue().getSizeEstimate()) * -1);
+    }
+
+    private void updateCurrentDataSizeInCache(long expectedSize, int delta) {
+        currentDataSizeInCache.compareAndSet(expectedSize, expectedSize + delta);
+    }
+
+    private double calculateEstimatedAvgEntrySize() {
+        if (readCache.size() == 0) {
+            return 0;
+        }
+        return currentDataSizeInCache.get() / readCache.size();
```

Comment:
Integer division in floating-point context, cast one of the operands to a double

Reply:
done